### ISSUE TITLE
Suggestions for implementation

### ISSUE THREAD

**Issue by rossmarkflo** [Ylianst/MeshCentral#2954]
The following would be nice to have:

- enforce that users use a login token - some backdoor accounts might need to be able to bypass that need
- alternative let the admin set a flag per user to make sure they enable 2FA/MFA
- list view for computers - add the 3x columns for AV / FW / UPDATE status for overview
- search for username in view for computers
- REST API - to begin with - for us we would love that our asset system can push out descriptions to MESH - depending on the hostname.. this way we can make this more useful.. a REST API would be perfect as we can engage some script to sync. this up
- pop-out remote desktop session - possibly even per monitor in to separate windows
- SHIFT key and cursors seem not to work in remote sessions to mark text - e.g. SHIFT + HOME or SHIFT + ARROW DOWN
- the blue indicator that someone is on the system - connected - should show WHO when hovering
- same if you are in the system - e.g. in GENERAL or somewhere else - should show someone else is remote connected already.. 

Might be we missed the options for some of those - sorry if so.. just wanted to provide a nice suggestions list. LOVE the project! Great work!

**Comment by Ylianst** [comment]
In general, I would prefer opening a separate issue for each request. This way, they can independently be tracked.

You can mandate 2FA now using the PasswordRequirements section that must be included in the domain section of the config.json. You can also skip 2FA if the connection is coming from some IP addresses.

```
      "PasswordRequirements": {
        "min": 8,
        "max": 64,
        "upper": 1,
        "lower": 1,
        "numeric": 1,
        "nonalpha": 1,
        "reset": 10,
        "hint": false,
        "force2factor": true,               <-----------------------
        "email2factor": true,
        "sms2factor": true,
        "push2factor": false,
        "oldPasswordBan": 5,
        "banCommonPasswords": true,
        "skip2factor": "127.0.0.1,192.168.2.0/24",     <----------
        "loginTokens": true,
        "twoFactorTimeout": 300,
        "autofido2fa": false
      },
```

To search for username, type `user:bob` or`u:bob` in the device search box.

For pop-out, host shift and click on a device. Or right click on a device, host shift and hit "desktop".

For keys that do not work in remote desktop, they are often blocked by the browser. You can right click on the custom keys and customize them as you like.

![image](https://user-images.githubusercontent.com/1319013/127388484-6216bf78-a758-4928-84e0-7683c6aadab5.png)

For the rest, open individual issues. Probably good to start with the high priority ones.

**Comment by rossmarkflo** [comment]
With the SHIFT key - what we found is - it is not fully transmitted at all. We can't do SHIFT + DEL either.. I mean you sure can add this to the send command list but for some other occasions this won't help.

We are talking about standard Windows browsers - not sure if you can provide some insight as of why the SHIFT won't be transmitted.. ?
